webapp/src/main/java/com/cmds/webapp/models/Robot.java

```diff
@@ -60,7 +70,14 @@ public void addTrip (Delivery delivery){
      * @return True if the trip was removed, false otherwise.
      */
     public boolean removeTrip (Long tripId) {
```

Comment:
Dont do the check here. this check should just immediately remove the trip without any checks.